pulpcore/cli/rpm/content.py

```diff
@@ -52,74 +65,215 @@ def _sha256_artifact_callback(
         required=True,
         help=_("Digest of the artifact to use"),
         callback=_sha256_artifact_callback,
+        allowed_with_contexts=(
+            PulpRpmPackageContext,
+            PulpRpmModulemdDefaultsContext,
+            PulpRpmModulemdContext,
+        ),
     ),
 ]
 
-
 content.add_command(list_command(decorators=list_options))
 content.add_command(show_command(decorators=lookup_options))
-content.add_command(create_command(decorators=create_options))
+content.add_command(
+    create_command(
+        decorators=create_options,
+        allowed_with_contexts=(
+            PulpRpmPackageContext,
+            PulpRpmModulemdDefaultsContext,
+            PulpRpmModulemdContext,
+        ),
+    )
+)
 
 
-@content.command()
-@click.option("--relative-path", required=True)
-@click.option("--file", type=click.File("rb"), required=True)
+# upload works for advisory/modulemd_default/modulemd/package
+# upload takes a file-argument and uploads to pulp, creates the artifact,
+# then creates the entity from it
+@content.command(
+    allowed_with_contexts=(
+        PulpRpmAdvisoryContext,
+        PulpRpmPackageContext,
+        PulpRpmModulemdDefaultsContext,
+        PulpRpmModulemdContext,
+    )
+)
 @chunk_size_option
-@pass_entity_context
+@pulp_option(
+    "--relative-path",
+    required=True,
+    help=_("Relative path within a distribution of the entity"),
+    allowed_with_contexts=(
+        PulpRpmPackageContext,
+        PulpRpmModulemdDefaultsContext,
+        PulpRpmModulemdContext,
+    ),
+)
+@pulp_option(
+    "--file",
+    type=click.File("rb"),
+    required=True,
+    help=_("An RPM binary"),
+    allowed_with_contexts=(PulpRpmPackageContext,),
+)
+@pulp_option(
+    "--file",
+    type=click.File("rb"),
+    required=True,
+    help=_("A modulemd/defaults JSON file"),
+    allowed_with_contexts=(PulpRpmModulemdDefaultsContext, PulpRpmModulemdContext),
+)
+@pulp_option(
+    "--file",
+    type=click.File("rb"),
+    required=True,
+    help=_("An advisory JSON file"),
+    allowed_with_contexts=(PulpRpmAdvisoryContext,),
+)
 @pass_pulp_context
+@pass_entity_context
 def upload(
     pulp_ctx: PulpContext,
-    entity_ctx: PulpRpmPackageContext,
-    relative_path: str,
-    file: IO[bytes],
-    chunk_size: int,
+    entity_ctx: Union[
+        PulpRpmAdvisoryContext,
+        PulpRpmPackageContext,
+        PulpRpmModulemdDefaultsContext,
+        PulpRpmModulemdContext,
+    ],
+    **kwargs
 ) -> None:
-    """Create an rpm package content unit by uploading a file"""
-    artifact_href = PulpArtifactContext(pulp_ctx).upload(file, chunk_size)
-    content = {"relative_path": relative_path, "artifact": artifact_href}
-    result = entity_ctx.create(body=content)
+    """Create an rpm content unit by uploading a file"""
+    file = kwargs.pop("file")
+    chunk_size = kwargs.pop("chunk_size")
+    artifact_ctx = PulpArtifactContext(pulp_ctx)
+
+    artifact_href = artifact_ctx.upload(file, chunk_size)
```

Comment:
pulp_ctx is a PulpRpmAdvisoryContext, whose ID_PREFIX is added to artifacts_list and fails by trying to find content_rpm_advisories_artifacts_list